atlasdb-tests-shared/src/test/java/com/palantir/atlasdb/transaction/impl/AbstractSnapshotTransactionTest.java

```diff
@@ -441,6 +448,36 @@ public void testLockAfterGet() throws Exception {
         });
 
         PathTypeTracker pathTypeTracker = PathTypeTrackers.constructSynchronousTracker();
+        DeleteExecutor typedDeleteExecutor = new DefaultDeleteExecutor(keyValueService, deleteExecutor);
+        KeyValueSnapshotReaderManager manager = new DefaultKeyValueSnapshotReaderManager(
+                new TransactionKeyValueServiceManager() {
+                    @Override
+                    public TransactionKeyValueService getTransactionKeyValueService(LongSupplier timestampSupplier) {
+                        return tkvsMock;
+                    }
+
+                    @Override
+                    public Optional<KeyValueService> getKeyValueService() {
+                        return Optional.empty();
+                    }
+
+                    @Override
+                    public DdlManager getDdlManager() {
+                        throw new SafeIllegalStateException("Not expecting to call in to DDL manager here");
+                    }
+
+                    @Override
+                    public boolean isInitialized() {
+                        return true;
+                    }
+
+                    @Override
+                    public void close() {}
+                },
```

Comment:
we might want to have a dedicated test double class for this kind of stuff, I expect it'll be pretty annoying to have to change one of these inline as/when we might add functionality to KVSRManager

Reply:
done